addons/paint-snap/ui.js

```diff
@@ -77,6 +77,10 @@ export function initUI({ addon, msg }) {
   settingButton.appendChild(createButtonImage("settings"));
   controlsGroup.appendChild(settingButton);
 
+  document.body.addEventListener("click", (e) => {
+    if (areSettingsOpen() && !e.target.matches("[class*=sa-paint-snap-]")) setSettingsOpen(false);
```

Comment:
A cleaner way to match any elements inside the settings panel, without taking advantage of this class name pattern (which might break in the future) is the following:

```suggestion
    if (areSettingsOpen() && !e.target.matches(".sa-paint-snap-settings *")) setSettingsOpen(false);
```

Reply:
The red one might break in the future, the green one is broken in the present. 😭

Reply:
Any ideas on how we could find a compromise?

Reply:
`!e.target.matches(".sa-paint-snap-settings *, .sa-paint-snap-button *"))`